cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/TimePartitionedStreamFileWriter.java

```diff
@@ -64,8 +69,6 @@
   private final long partitionDuration;
   private TimePartition timePartition = new TimePartition(-1L);
 
-  // TODO: Add a timer task to close file after duration has passed even there is no writer.
```

Comment:
I'm not sure what it means by "even there is no writer".
There will always be a writer, I think. There just may be no new events.